app/controllers/sessions/invite_to_clinic_controller.rb

```diff
@@ -44,36 +50,51 @@ def set_session
 
   def set_generic_clinic_session
     @generic_clinic_session =
-      (
-        if @session.clinic?
-          @session
-        else
-          @session.team.generic_clinic_session(
-            academic_year: @session.academic_year
-          )
-        end
-      )
+      if @session.clinic?
+        @session
+      else
+        @session.team.generic_clinic_session(
+          academic_year: @session.academic_year
+        )
+      end
   end
 
-  def set_invitations_to_send
+  def set_patient_sessions_to_invite
     session_date = @generic_clinic_session.next_date(include_today: true)
 
-    @invitations_to_send =
-      if @session.school?
-        SendClinicInitialInvitationsJob
-          .new
-          .patient_sessions(
-            @generic_clinic_session,
-            school: @session.location,
-            programmes: @session.programmes.to_a,
-            session_date:
+    if @session.school?
+      patient_sessions_in_school = @session.patient_sessions.includes(:patient)
+
+      patient_sessions_in_clinic =
+        patient_sessions_in_school.map do |patient_session|
+          PatientSession.find_or_initialize_by(
+            patient: patient_session.patient,
+            session: @generic_clinic_session
           )
-          .length
-      else
-        SendClinicSubsequentInvitationsJob
-          .new
-          .patient_sessions(@session, session_date:)
-          .length
-      end
+        end
+
+      programmes = @session.programmes.to_a
+
+      @patient_sessions_to_invite =
+        patient_sessions_in_clinic
+          .reject { it.session_notifications.any? }
```

Comment:
Is it worth including these with the original request (line 66)?

Reply:
I must be missing something ... did this not trigger a `strict_loading` error?

Reply:
I think it's because they're new objects. There is an N+1 issue here, but I can add an `includes` to cover that.